lib/gtfs/cache_file.ex

```diff
@@ -0,0 +1,127 @@
+defmodule Gtfs.CacheFile do
+  @moduledoc """
+  This module holds the logic for loading and saving cache files
+  in dev and test environments with the goal of decreasing startup
+  time so devs can work more quickly.
+
+  If GTFS data changes, the cache files should be deleted/removed
+  to allowing caching of the changes.
+  """
+  require Logger
+
+  alias Gtfs.Helpers.EnvHelper
+
+  @directory Path.join([File.cwd!(), "priv/gtfs_cache"])
+
+  @dev_filename "dev_cache.terms"
+  @test_filename "test_cache.terms"
+
+  @doc """
+  The application should use the file to load state for GTFS
+  if the the envs are :dev or :test.
+
+    iex> Mix.env
+    :test
+    iex> CacheFile.should_use_file?
+    true
+  """
+  @spec should_use_file?() :: boolean
+  def should_use_file? do
+    EnvHelper.is_env?(:dev) || EnvHelper.is_env?(:test)
```

Comment:
This might not work, and it might not be a good idea, but what if we could decide whether to use the cache from config instead of by checking the environment, by having a 3rd file source `{:cached_url, url}`?

Cons:
* Depends on my other comment about moving the cache check into the fetch step.
* Requires specifying a `files_source()` in the config instead of just a `String.t()` url. (This is even harder for specifying the source be environment variable, but perhaps we could just always use the uncached `files_source` in that case.)
* Makes dev+prod more different.
* Makes bad states like trying to use cached data on prod more possible by uncoupling the decision of whether to use the cache from the environment.

Pros:
* Doesn't need the `EnvHelper` and other infrastructure.
* The decision about whether to use the cache or not becomes simpler.
* More testable because it make decisions based on the argument instead of global state.

Reply:
Ah, hadn't seen Paul's comments.

If the config has `cache_filename` or `use_cache?`, it'd be much easier to create a `files_source()` based on that.